src/framework/theme/components/button-group/button-toggle.directive.ts

```diff
@@ -123,6 +125,9 @@ export class NbButtonToggleDirective extends NbButton {
 
   @HostListener('click')
   onClick(): void {
+    if (!this.buttonGroup?.multiple && this.pressed) {
+      return;
+    }
     this.pressed = !this.pressed;
```

Comment:
Lets add a comment clarifying the case when we need to prevent pressed state change. And what do you think about changing to:
```suggestion
    if (this.buttonGroup?.multiple || !this.pressed) {
      this.pressed = !this.pressed;
    }
```